src/app/query/bitcoin/ordinals/brc20/brc20-tokens.hooks.ts

```diff
@@ -92,18 +96,19 @@ function makeBrc20Token(token: Brc20Token) {
     marketData: token.min_listed_unit_price
       ? createMarketData(
           createMarketPair(token.ticker, 'USD'),
-          createMoney(new BigNumber(token.min_listed_unit_price), 'USD')
+          createMoney(new BigNumber(token.min_listed_unit_price).times(satPrice), 'USD')
         )
       : null,
   };
 }
 
 export function useBrc20Tokens() {
+  const btcMarketData = useCryptoCurrencyMarketDataMeanAverage('BTC');
   const { data: allBrc20TokensResponse } = useGetBrc20TokensQuery();
   const brc20Tokens = allBrc20TokensResponse?.pages
     .flatMap(page => page.brc20Tokens)
     .filter(token => token.length > 0)
     .flatMap(token => token);
 
-  return brc20Tokens?.map(token => makeBrc20Token(token)) ?? [];
+  return brc20Tokens?.map(token => makeBrc20Token(btcMarketData, token)) ?? [];
```

Comment:
I see that you're just fixing the existing implementation, but I can't help but feel it's a bit off to pair together the models that 1) represent a token, and 2) contains marketdata.

For example, `Money` shouldn't contain the USD exchange data, that belongs in `MarketData`, whereas here they're merged. 

Refactor not completely necessary, but maybe something to think about

Reply:
Another thought on this, we already have all the methods to convert to fiat, so why not use those in the components where you need the data, and handle the types separately, rather than merging?

Reply:
I was struggling with the best way to do this quickly bc thought we would want the fix out. If we don't thou, I can dive back into it today to see if I can come up with something better. Some of the things I was trying initially weren't working as I expected, so will try again today.